nimble/host/audio/services/bass/src/ble_audio_svc_bass.c

```diff
@@ -298,10 +320,23 @@ ble_svc_audio_bass_add_source(uint8_t *data, uint16_t data_len, uint16_t conn_ha
     operation.op = BLE_SVC_AUDIO_BASS_OPERATION_ADD_SOURCE;
     operation.conn_handle = conn_handle;
 
+    offset++;
     operation.add_source.adv_addr.type = data[offset++];
+    if (operation.add_source.adv_addr.type < 0 ||
+        operation.add_source.adv_addr.type > 3) {
```

Comment:
type won't be < 0 as it's uint8. I wonder whether we need such check in this place. I there any test that test for such scenario? Maybe we could have a assert check defined globally in ble.h so that you could use it in this place.

Reply:
There is a test that ensures we detect those scenarios. 
![image](https://github.com/user-attachments/assets/79641985-903f-4283-a8e1-69d6b0e7d287)

Reply:
Please check if current check looks good. :)